### ISSUE TITLE
Integration with pytorch_geometry?

### ISSUE THREAD

**Issue by ninamiolane** [geomstats/geomstats#840]
Pytorch geometric is a great package for geometric deep learning, focused on learning on arbitrary graphs as well as on 3D meshes or point clouds. We could think on how we can interact with it.

https://github.com/rusty1s/pytorch_geometric

**Comment by Bawaw** [comment]
I would love to see the torch geometric Data object being incorporated in geomstats as it would allow us to work with differently shaped discretizations and work with larger datasets! 

e.g.

```
import os
os.environ['GEOMSTATS_BACKEND'] = 'pytorch'

import torch
import random
from geomstats.learning.frechet_mean import FrechetMean
from geomstats.geometry.riemannian_metric import RiemannianMetric
from torch_geometric.data import Data, InMemoryDataset

B = 5

# pointcloud/mesh dataset
class PCDataset(InMemoryDataset):
    def __init__(self):
        super().__init__()
        pcs = [torch.rand(random.randint(1, 100), 3) for _ in range(B)]
        prepped_pcs = [Data(pos=pcs[i]) for i in range(B)]
        self.data, self.slices = InMemoryDataset.collate(prepped_pcs)
pc_dataset = PCDataset()


some_metric = RiemannianMetric(dim=pc_dataset[0].pos.view(-1).shape[0])
mean = FrechetMean(some_metric, point_type='tg_data')
breakpoint()
mean.fit(pc_dataset)
```

Best, 
Balder

**Comment by ninamiolane** [comment]
Great point, thanks for sharing that use case! No one is actively working on this integration at the moment, but we will let you know if that changes. Unless @nguigs you know more about this?

**Comment by nguigs** [comment]
Thanks for the example! I haven't had the opportunity to look at it yet, but this is planned! What metric did you have in mind for point clounds? And for meshes, besides LDDMM (#848) ?

**Comment by Bawaw** [comment]
So my research is on finding ways to make computational anatomy feasible on large datasets (>10k high resolution data e.g. MRI, CBCT, surface scans, ...) to do this I'm mostly looking at parametrising registration models using (geometric) deep learning. So far I have only looked at the SVF and LDDMM framework as shape metrics. However, the goal is that these models will be used in our/the lab for doing statistical analysis (e.g. for craniofacial reconstruction or for heritability studies) and for that I would like to interface with Geomstats. I've made a list of some features I would like to implement/use in #848. 

Let me know if there is something I can do to help, I'm not a mathematician but I do have a CS background. 

Best regards, 
Balder
